### PR TITLE
Removed inline js from breadcrumb_select.html.

### LINKED ISSUE #8380: Remove inline js: books/breadcrumb_select.html
Remove the inline JS from this file following the recommended steps in #4474

### Proposal & Constraints

This should be relatively easy to make and test.

### Related files

https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/books/breadcrumb_select.html

### PR DESCRIPTION
**Pull request by imperial-chief**
Added and improved functionality of inline javascript in breadcrumb_select.js file. Imported this breadcrumb_select.js file in index.js.

Closes #8380 Remove inline js: books/breadcrumb_select.html

Refactor

### Technical

Removed inline js from breadcrumb_select.html.
Updated index.js and created a file openlibrary/plugins/openlibrary/js/breadcrumb_select/index.js for using the inline js functionality.

### Testing
1. Visit http://localhost/account/books/currently-reading
2. In the select options available try pressing the keys 'ArrowUp', 'ArrowDown' to prevent changes.
3. In the select options available try pressing the keys 'Tab', 'Enter', ' '  to allow changes to take place.

### Screenshot
Screenshot from 2023-10-24 18-37-31
[Screencast from 2023-10-24 18-37-44.webm](https://github.com/internetarchive/openlibrary/assets/117163483/b0aa0d3f-03c4-4de9-88fa-2f8359693167)

### Stakeholders

@RayBB

### CODE REVIEW

**Comment by imperial-chief** [comment]
@RayBB any changes?

**Comment by RayBB** [comment]
@imperial-chief can you please provide instructions for testing and screenshots of it working?

**Comment by RayBB** [comment]
The testing should be go to a page like "currently reading" and then use dropdown to move to another page.
Also, I think we should add something like [this](https://github.com/internetarchive/openlibrary/pull/8374/files#diff-587fb865c85bb3325504c795a40db189d2e702d42d9d160304607e027c142ebd) but with `document.querySelectorAll('.crumb select');`

Overall it looks good though!

**Comment by imperial-chief** [comment]
Okay i will try doing so.

> @imperial-chief can you please provide instructions for testing and screenshots of it working?
What do you mean by 'please provide instructions for testing'

**Comment by RayBB** [comment]
Someone else who isn't me will test the PR. We should write steps for them to test it so they don't need to figure out what page the code runs on.

**Comment by imperial-chief** [comment]
@RayBB i tried finding the usage of this,
But i could find exactly where this breadcrumbs are used.
Can you help me find it?

**Comment by imperial-chief** [comment]
@RayBB let me know for any changes further.

**Comment by RayBB** [comment]
@imperial-chief the code looks good.
Can you please test that it works and provide screenshots and testing instructions.
The breakcrumbs are found on a page like https://openlibrary.org/account/books/currently-reading 
In this dropdown:
image

**Comment by imperial-chief** [comment]
@RayBB i tried running the changes but i failed.
Not only that, while debugging i tried to find where was my code breaking and found that index.js wasn't getting executed at all (i added many `console.log()` statements in complete index.js but i wasn't able to see any of them).
Can you please help me out with this?

**Comment by RayBB** [comment]
@imperial-chief did you try rebuilding the JS?
This can be done by running: `docker compose run --rm home make js`

Or `docker-compose run --rm home npm run-script watch` which will watch for changes to the JS file and recompile after each change.

A bit more info on docker here: https://github.com/internetarchive/openlibrary/blob/master/docker/README.md

If you're still having trouble you can watch some of the onboarding videos at https://github.com/internetarchive/openlibrary/blob/master/CONTRIBUTING.md

**Comment by imperial-chief** [comment]
@RayBB i uploaded a screenshot and a screencast link of testing.
I also updated the testing instructions in the PR main comment.

**Comment by RayBB** [comment]
Looks good to me. Now we just need @jimchamp to review and merge :)

**Comment by imperial-chief** [comment]
Sure!

File: openlibrary/templates/books/breadcrumb_select.html [hunk 1]
```diff
@@ -40,29 +40,4 @@ <h2 class="breadcrumb-title">$breadcrumb_title</h2>
...
-
-            nav.addEventListener('keydown', ({ key }) => {
-                if (preventedKeys.has(key))
-                ignoreChange = true
-                else if (allowedKeys.has(key))
-                ignoreChange = false
-            })
-            })
-        </script>
+        </span>
```
Comment (jimchamp) [inline openlibrary/templates/books/breadcrumb_select.html:43]:
```suggestion
        </span>

```
Add newline to end of file.

File: openlibrary/plugins/openlibrary/js/breadcrumb_select/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+export function initBreadcrumbSelect(crumbs) {
+    const allowedKeys = new Set(['Tab', 'Enter', ' ']);
+    const preventedKeys = new Set(['ArrowUp', 'ArrowDown']);
+
+    function handleNavEvents(nav) {
+        let ignoreChange = false;
+
+        nav.addEventListener('change', () => {
+            if (ignoreChange) return;
+            // It's actually changed!
```
Comment (jimchamp) [inline openlibrary/plugins/openlibrary/js/breadcrumb_select/index.js:15]:
```suggestion
    // watch crumbs for changes,
    // ensures it's a full value change, not a user exploring options via keyboard
    function handleNavEvents(nav) {
        let ignoreChange = false;

        nav.addEventListener('change', () => {
            if (ignoreChange) return;
```
Keep important comments, and remove frivolous comment.

**Review by jimchamp — APPROVED** [review]
Thanks @imperial-chief!  Will merge after I commit my suggested changes.

**Pull request merged**